Sources/StreamChat/Repositories/MessageRepository.swift

```diff
@@ -56,8 +62,8 @@ class MessageRepository {
             }, completion: { error in
                 if let error = error {
                     log.error("Error changing localMessageState message with id \(messageId) to `sending`: \(error)")
-                    self?.markMessageAsFailedToSend(id: messageId) {
-                        completion(.failure(.failedToSendMessage))
+                    self?.markMessageAsFailedToSend(id: messageId, isForced: true) { error in
+                        completion(.failure(error == nil ? .failedToSendMessage : .failedToSendMessageWithStatusUpdateError))
```

Comment:
Even with the above fix, we could still fail to mark the message's status to `.sendingFailed` due to DB error, again.
In that case, report `failedToSendMessageWithStatusUpdateError` (added in this PR) instead of `failedToSendMessage`, so that we can retry sending rather than silently fail and ignore resulting message reordering